### ISSUE TITLE
Credo diff returns non-0 exit codes for issues not in the diff.

### ISSUE THREAD

**Issue by Ch4s3** [rrrene/credo#837]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`1.5.2`):
* Erlang/Elixir version (`1.10.4-otp-23`):
* Operating system: ubuntu and MacOS

### What were you trying to do?

I am trying to introduce a check on adding `@spec` but there are 1597 old functions without specs. Obviously I can't go back and update all of them in one shot but would like to make all new PRs use specs.

### Expected outcome
If `mix credo diff master` returns `+  added no issues,` I would expect an exit code of 0.

### Actual outcome

```sh
#!/bin/bash -eo pipefail -l
mix credo diff master --config-name default -a
Diffing 629 source files in working dir with master (this might take a while) ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 6.5 seconds (1.1s to load, 5.4s running 58 checks on 629 files)

Changes between master and working dir:

+  added no issues,
✔  fixed no issues, and
~  kept 1597 code readability issues, 1 software design suggestion.

Use `mix credo explain` to explain issues, `mix credo diff --help` for options.

Exited with code exit status 4
CircleCI received exit code 4
```

**Comment by Ch4s3** [comment]
FWIW I was able to reproduce this behavior on `1.5.3` as well.

**Comment by rrrene** [comment]
Thanks for reporting this :+1: I should release a hotfix, so it is now fixed on the 1.5.x branch (`release/1.5`).

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo", branch: "release/1.5"}
```

Please report back if your issue is solved! 👍

**Comment by Ch4s3** [comment]
This looks good mostly, but it seems like some duplicate errors are still firing.

1. I added this branch to my mix file and compiled. in a branch called `credo-1.5.x`
2. Then I re-added some checks to for example our test files. 
3. Then I committed this and checked out a new branch `git checkout -b test`
4. Then I ran `mix credo diff credo-1.5.x --config-name default -a`

I then see the following:
```bash
Analysis took 10.8 seconds (1.3s to load, 9.5s running 58 checks on 943 files)

Changes between credo-1.5.x and working dir:

+  added 11 new software design suggestions,
✔  fixed 11 software design suggestions, and
~  kept 17 consistency issues, 12 warnings, 98 refactoring opportunities, 1934 code readability issues, 293 software design suggestions.
```

**Comment by Ch4s3** [comment]
If I don't check tests It works as expected.

**Comment by rrrene** [comment]
I pushed a possible fix to the `release/1.5` branch.

```elixir
{:credo, github: "rrrene/credo", branch: "release/1.5"}
```

Can you try again?

**Comment by Ch4s3** [comment]
Hey, sorry for the delay. This worked.
